streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -688,6 +690,8 @@ class StreamingContext private[streaming] (
     } finally {
       // The state should always be Stopped after calling `stop()`, even if we haven't started yet
       state = STOPPED
+      // De-registering Streaming Metrics of the StreamingContext
+      env.metricsSystem.removeSource(streamingSource)
```

Comment:
I think it would be better to add try as TD said, otherwise you may meet exception when SparkContext is stopped.